cogeo_mosaic/overviews.py

```diff
@@ -124,80 +128,67 @@ def create_low_level_cogs(
-                                )
-
-                            return window, tile, mask
-
-                        with futures.ThreadPoolExecutor(
-                            max_workers=threads
-                        ) as executor:
-                            future_work = [
-                                executor.submit(_get_tile, item) for item in blocks
-                            ]
-                            with click.progressbar(
-                                futures.as_completed(future_work),
-                                length=len(future_work),
-                                show_percent=True,
-                            ) as future:
-                                for res in future:
-                                    pass
-
-                        for f in _filter_futures(future_work):
-                            window, tile, mask = f
+                # with rasterio_open(in_memory, out_path, **params) as mem:
+                with rasterio.open(out_path, "w", **params) as dst:
+
+                    def _get_tile(wind):
+                        idx, window = wind
+                        x = extrema["x"]["min"] + idx[1]
+                        y = extrema["y"]["min"] + idx[0]
+                        t = mercantile.Tile(x, y, base_zoom)
+
+                        kds = set(find_quadkeys(t, mosaic.quadkey_zoom))
+                        if not mosaic_quadkeys.intersection(kds):
+                            return window, None, None
+
+                        assets = mosaic.tile(*t)
+                        if not assets:
+                            return window, None, None
+
+                        tile, mask = mosaic_tiler(
+                            assets,
+                            x,
+                            y,
+                            base_zoom,
+                            cogeo.tile,
+                            indexes=info["indexes"],
+                            tilesize=tilesize,
+                            pixel_selection=defaults.FirstMethod(),
+                        )
+                        return window, tile, mask
+
+                    with futures.ThreadPoolExecutor(max_workers=threads) as executor:
+                        future_work = executor.map(_get_tile, blocks)
+                        for result in tqdm(future_work, total=len(blocks)):
+                            window, tile, mask = result
                             if tile is None:
                                 continue
 
-                            mem.write(tile, window=window)
-                            if info[5]:
-                                mem.write_mask(mask.astype("uint8"), window=window)
+                            dst.write(tile, window=window)
+                            if info["mask"]:
+                                dst.write_mask(mask.astype("uint8"), window=window)
 
+                    # Not already written to a file
+                    if in_memory:
```

Comment:
so if you don't use `in_memory` you don't create a COG? 👎

Reply:
https://github.com/developmentseed/cogeo-mosaic/pull/91#discussion_r453224648

Reply:
well you can't create a COG directly you need to perform a `copy` (which is what rio-cogeo does)

Reply:
I understand; ideally it would be nice to support a low-memory option, where you first write a geotiff using windows and then on the second pass you call `rio-cogeo` on that geotiff with a low-memory option there too.